### PR TITLE
[Core] Refactor executor classes to make it easier to inherit GPUExecutor

### PR DESCRIPTION
**Pull request by jikunshang**
FILL IN THE PR DESCRIPTION HERE

XPUExecutor could/should share most code from GPUExecutor, so refactor `_get_create_worker_kwargs` method and `update_environment_variables` related logic to make it easier to inherit.

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to vLLM! Before submitting the pull request, please ensure the PR meets the following criteria. This helps vLLM maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Frontend]</code> For changes on the vLLM frontend (e.g., OpenAI API server, <code>LLM</code> class, etc.) </li>
    <li><code>[Kernel]</code> for changes affecting CUDA kernels or other compute kernels.</li>
    <li><code>[Core]</code> for changes in the core vLLM logic (e.g., <code>LLMEngine</code>, <code>AsyncLLMEngine</code>, <code>Scheduler</code>, etc.)</li>
    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix (e.g., <code>[Hardware][AMD]</code>).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
    <li>Pass all linter checks. Please use <a href="https://github.com/vllm-project/vllm/blob/main/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of vLLM. It helps vLLM user understand and utilize the new features or changes.</li>
</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes (>500 LOC excluding kernel/data/config/test), we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the vLLM team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the vLLM team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the vLLM team.</li>
    <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>
    <li> Please respond to all comments within a reasonable time frame. If a comment isn't clear or you disagree with a suggestion, feel free to ask for clarification or discuss the suggestion.
 </li>
</ul>

<h3>Thank You</h3>

<p> Finally, thank you for taking the time to read these guidelines and for your interest in contributing to vLLM. Your contributions make vLLM a great tool for everyone! </p>

</details>

### CODE REVIEW

**Comment by jikunshang** [comment]
@youkaichao Please take a review, thanks.

File: vllm/executor/gpu_executor.py
Comment (youkaichao) [inline vllm/executor/gpu_executor.py:1]: this file change makes sense to me.

File: vllm/executor/ray_gpu_executor.py [hunk 1]
```diff
@@ -91,22 +91,48 @@ def _configure_ray_workers_use_nsight(self,
...
-            worker_module_name = "vllm.spec_decode.spec_decode_worker"
-            worker_class_name = "create_spec_worker"
-        elif self.scheduler_config.is_multi_step:
-            worker_module_name = "vllm.worker.multi_step_worker"
-            worker_class_name = "MultiStepWorker"
-        else:
-            worker_module_name = "vllm.worker.worker"
-            worker_class_name = "Worker"
+        (worker_module_name,
+         worker_class_name) = self._get_worker_module_and_class()
```
Comment (youkaichao) [inline vllm/executor/ray_gpu_executor.py:95]: this change makes sense to me.

File: vllm/executor/ray_gpu_executor.py [hunk 2]
```diff
@@ -91,22 +91,48 @@ def _configure_ray_workers_use_nsight(self,
...
+        all_args_to_update_environment_variables = [({
+            "CUDA_VISIBLE_DEVICES":
+            ",".join(map(str, node_gpus[node_id])),
+            "VLLM_INSTANCE_ID":
+            VLLM_INSTANCE_ID,
+            "VLLM_TRACE_FUNCTION":
+            str(envs.VLLM_TRACE_FUNCTION),
+        }, ) for (node_id, _) in worker_node_and_gpu_ids]
+        return all_args_to_update_environment_variables
+
```
Comment (youkaichao) [inline vllm/executor/ray_gpu_executor.py:135]: this looks unnecessary to me. I believe you copied quite a lot lines of code here.

File: vllm/executor/ray_gpu_executor.py [hunk 3]
```diff
@@ -220,19 +246,8 @@ def sort_by_driver_then_worker_ip(worker):
...
-        # Set environment variables for the driver and workers.
-        all_args_to_update_environment_variables = [({
-            "CUDA_VISIBLE_DEVICES":
-            ",".join(map(str, node_gpus[node_id])),
-            "VLLM_INSTANCE_ID":
-            VLLM_INSTANCE_ID,
-            "VLLM_TRACE_FUNCTION":
-            str(envs.VLLM_TRACE_FUNCTION),
-        }, ) for (node_id, _) in worker_node_and_gpu_ids]
         self._run_workers("update_environment_variables",
```
Comment (youkaichao) [inline vllm/executor/ray_gpu_executor.py:249]: I think you can add one line here:

```python
self._env_vars_for_all_workers = all_args_to_update_environment_variables
```

And implement `_get_env_vars_to_be_updated` as directly returning `self._env_vars_for_all_workers`.

If you want to add new env vars later via inheritance, you can change the implementation of `_get_env_vars_to_be_updated` .

**Comment by youkaichao** [comment]
@jikunshang thanks for your kind consideration! This PR is indeed small enough such that I can review directly ❤️

File: vllm/executor/ray_gpu_executor.py (hunk 3 above)
Comment (jikunshang) [reply vllm/executor/ray_gpu_executor.py:234]: good idea! changed as your suggestion.

**Review by youkaichao — APPROVED** [review]
Looks much cleaner now!

**Comment by youkaichao** [comment]
please fix the format btw

**Comment by jikunshang** [comment]
> please fix the format btw

Done, thanks.

**Pull request merged**